pkg/session_recording/recordings.jsx

```diff
@@ -529,7 +531,13 @@
         }
 
         componentDidMount() {
-            this.journalctlStart();
+            let proc = cockpit.spawn(["getent", "passwd", "tlog"]);
+
+            proc.stream((data) => {
+                this.uid = data.split(":",3)[2];
+                this.journalctlStart();
```

Comment:
From cockpit.js:
`        var ret = dfd.promise;
        ret.stream = function(callback) {
            buffer.callback = callback.bind(ret);
            return this;
        };

        ret.input = function(message, stream) {
            if (message !== null && message !== undefined) {
                spawn_debug("process input:", message);
                channel.send(message);
            }
            if (!stream)
                channel.control({ command: "done" });
            return this;
        };

        ret.close = function(problem) {
            spawn_debug("process closing:", problem);
            if (channel.valid)
                channel.close(problem);
            return this;
        };

        return ret;`

So, .input expects stream as a second parameter. I think I can keep .stream but I'll add .close() in a callback.